crates/miden-tx/src/prover/mod.rs

```diff
@@ -85,10 +96,22 @@ impl LocalTransactionProver {
             .map_err(TransactionProverError::RemoveFeeAssetFromDelta)?;
 
         // If the account is on-chain, add the update details.
-        let builder = match account.is_onchain() {
+        let builder = match partial_account.id().is_onchain() {
             true => {
-                let account_update_details = if account.is_new() {
-                    let mut account = account.clone();
+                let account_update_details = if partial_account.is_new() {
+                    let (id, nonce, code, partial_storage, partial_vault) =
+                        partial_account.into_parts();
+
+                    // For new accounts, the partial storage must represent the full initial account
+                    // storage.
+                    let storage = partial_storage_to_full(partial_storage);
+
+                    // The vault of a new account should be empty.
+                    debug_assert_eq!(partial_vault.leaves().count(), 0);
+                    let vault = AssetVault::default();
+
+                    let mut account = Account::from_parts(id, vault, storage, code, nonce);
```

Comment:
Should we have `TryFrom<PartialAccount> for Account` implementation instead? Seems like it is more natural to have this code next to `PartialAccount`.

Reply:
I had this at first, but figured it would be removed again in the future, in which case it's easier to have it in one place (so we don't accidentally leave, say, `impl From<PartialStorageMap> for StorageMap>`).

Long-term, it's probably rarely going to be correct to convert a `PartialAccount` to an `Account`, `PartialStorage` to `AccountStorage`, etc. 

I'd create a separate `partial_account_to_full` helper function to encapsulate it a bit better, but let me know if you still prefer having it as `TryFrom`, happy to change it.